### PR TITLE
Upgrade pydocstyle to 2.0

### PR DESCRIPTION
**Pull request by Vamshi99**
Closes https://github.com/coala/coala-bears/issues/1645

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/python/PyDocStyleBear.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create_arguments(self, filename, file, config_file,
         """
         args = (filename,)
         if pydocstyle_ignore and pydocstyle_select:
-            self.err('The arguments pydocstyle_select and pydocstyle_ignore '
```
Comment (meetmangukiya) [inline bears/python/PyDocStyleBear.py]: This change is not relevant I think..

File: bears/python/PyDocStyleBear.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
 
 
 @linter(executable='pydocstyle',
-        use_stdout=False,
```
Comment (jayvdb) [inline bears/python/PyDocStyleBear.py]: Could this be change to explicit `use_stdout=True,`

File: bears/python/PyDocStyleBear.py [hunk 3]
```diff
@@ -4,7 +4,6 @@
...
         use_stderr=True,
```
Comment (jayvdb) [inline bears/python/PyDocStyleBear.py]: have you tried setting this to `False`?

Comment (Vamshi99) [inline bears/python/PyDocStyleBear.py]: @jayvdb Yeah, I tried that. If it is changed we are getting error in the output.

File: bears/python/PyDocStyleBear.py (hunk 2 above)
Comment (Vamshi99) [inline bears/python/PyDocStyleBear.py]: Suee

**Comment by Vamshi99** [comment]
@jayvdb PTAL 😃

File: bears/python/PyDocStyleBear.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 
 @linter(executable='pydocstyle',
-        use_stdout=False,
+        use_stdout=True,
         use_stderr=True,
```
Comment (jayvdb) [inline bears/python/PyDocStyleBear.py]: Ordinarily, I would agree with you.

However, as this is a violent change of behaviour in pydocstyle, I'd prefer that it is an explicit `use_stdout=True` here.

Comment (jayvdb) [inline bears/python/PyDocStyleBear.py]: Have you tried setting `use_stderr=False` ?

Comment (Vamshi99) [inline bears/python/PyDocStyleBear.py]: @jayvdb Yeah I have done that, it gives error in output. It says bear returned none on execution, although it should.

**Comment by jayvdb** [comment]
ack df053f7

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**